### ISSUE TITLE
packaging scripts/compile does not work on amazon linux 2023

### ISSUE THREAD

**Issue by EeroV** [libreswan/libreswan#1058]
due, that dns packages are not in amazon. there is workround to compile and package without dns support..

--- fedora/libreswan.spec       2023-03-26 17:31:17.306852178 +0000
+++ amazon-2023/libreswan.spec  2023-03-27 05:07:07.115022369 +0000
@@ -6,6 +6,7 @@
 %global nss_version 3.52
 %global unbound_version 1.6.6
 # Libreswan config options
+#/usr/libexec/ipsec/addconn
 %global libreswan_config \\\
     LIBEXECDIR=%{_libexecdir}/ipsec \\\
     MANDIR=%{_mandir} \\\
@@ -13,7 +14,7 @@
     INITSYSTEM=systemd \\\
     PYTHON_BINARY=%{__python3} \\\
     SHELL_BINARY=%{_bindir}/sh \\\
-    USE_DNSSEC=true \\\
+    USE_DNSSEC=false \\\
     USE_LABELED_IPSEC=true \\\
     USE_LDAP=true \\\
     USE_LIBCAP_NG=true \\\
@@ -47,7 +48,7 @@
 BuildRequires: flex
 BuildRequires: gcc
 BuildRequires: hostname
-BuildRequires: ldns-devel
+#BuildRequires: ldns-devel
 BuildRequires: libcap-ng-devel
 BuildRequires: libevent-devel
 BuildRequires: libseccomp-devel
@@ -101,6 +102,8 @@
 %setup -q -n libreswan-%{version}%{?prever}
 # enable crypto-policies support
 sed -i "s:#[ ]*include \(.*\)\(/crypto-policies/back-ends/libreswan.config\)$:include \1\2:" configs/ipsec.conf.in
+# fix path
+sed -i "s#/local##g" OBJ.linux.x86_64/initsystems/systemd/ipsec.service

 %build
 %make_build \

**Comment by cagney** [comment]
Can you post a diff of `OBJ.linux.x86_64/initsystems/systemd/ipsec.service`?  I can see that `/local` is being stripped out, but I'm not sure where.

**Comment by EeroV** [comment]
it strips out "local" part from /usr/local/libexec/ipsec/addconn

for some reason that macro expands to /usr/local/libexec/ and not
/usr//libexec/ -- not sure why? maybe there is better way to do this?

Eero

ma 27. maalisk. 2023 klo 15.34 cagney ***@***.***) kirjoitti:

**Comment by cagney** [comment]
which version of libreswan?
My guess is that this, from mk/config.mk, is somehow screwing up:
```
# "PREFIX" part of tree, used in building other pathnames.
PREFIX ?= /usr/local

# LIBEXECDIR is where sub-commands get put.  the "ipsec" command will
# look for them when it is run.
LIBEXECDIR ?= $(PREFIX)/libexec/ipsec
TRANSFORMS += 's:@@LIBEXECDIR@@:$(LIBEXECDIR):g'
TRANSFORMS += 's:@IPSEC_EXECDIR@:$(LIBEXECDIR):g'
USERLAND_CFLAGS += -DIPSEC_EXECDIR=\"$(LIBEXECDIR)\"
```

**Comment by EeroV** [comment]
that probably affects on other linuxes too? not amazon specific..

**Comment by bleve** [comment]
No, that is something new. As amazon linux seem to be another rhel clone, there should be extra repository providing ldns-devel etc extra development packages, just like there is in rhel/centos-stream

**Comment by EeroV** [comment]
Well, amazon linux is not based on rhel.

https://docs.aws.amazon.com/linux/al2023/ug/relationship-to-fedora.html


Eero

**Comment by bleve** [comment]
Can you make build log available?

**Comment by EeroV** [comment]
yes, just let me know, which files you need.

Eero

**Comment by bleve** [comment]
How do you run your build?

**Comment by EeroV** [comment]
rpmbuild -ba specfile..

Eero


ti 28. maalisk. 2023 klo 10.50 Tuomo Soini ***@***.***)
kirjoitti:

**Comment by bleve** [comment]
ok. Something like this would work:

rpmbuild -ba specfile 2>&1 | tee -a build.log

**Comment by cagney** [comment]
assuming pushed fix worked
